StyleCop.Analyzers/StyleCop.Analyzers.CodeFixes/ReadabilityRules/SA1130CodeFixProvider.cs

```diff
@@ -0,0 +1,163 @@
+        {
+            var parameterList = anonymousMethod.ParameterList;
+            SyntaxNode lambdaExpression;
+
+            if (parameterList == null)
+            {
+                parameterList = SyntaxFactory.ParameterList()
+                    .WithLeadingTrivia(anonymousMethod.DelegateKeyword.LeadingTrivia)
+                    .WithTrailingTrivia(anonymousMethod.DelegateKeyword.TrailingTrivia);
+            }
+            else
+            {
+                parameterList = parameterList.WithLeadingTrivia(anonymousMethod.DelegateKeyword.TrailingTrivia);
+            }
+
+            foreach (var parameter in parameterList.Parameters)
+            {
+                if (!IsValid(parameter))
+                {
+                    return anonymousMethod;
+                }
+            }
+
+            var arrowToken = SyntaxFactory.Token(SyntaxKind.EqualsGreaterThanToken)
+                .WithTrailingTrivia(SyntaxFactory.ElasticSpace);
+
+            if (parameterList.Parameters.Count == 1)
+            {
+                var parameterSyntax = RemoveType(parameterList.Parameters[0]);
+
+                var trailingTrivia = parameterSyntax.GetTrailingTrivia()
+                    .Concat(parameterList.CloseParenToken.LeadingTrivia)
+                    .Concat(parameterList.CloseParenToken.TrailingTrivia.WithoutTrailingWhitespace())
+                    .Concat(new[] { SyntaxFactory.ElasticSpace });
+                var leadingTrivia = parameterList.OpenParenToken.LeadingTrivia.Concat(parameterList.OpenParenToken.TrailingTrivia);
+
+                parameterSyntax = parameterSyntax.WithLeadingTrivia(leadingTrivia).WithTrailingTrivia(trailingTrivia);
+
+                lambdaExpression = SyntaxFactory.SimpleLambdaExpression(anonymousMethod.AsyncKeyword, parameterSyntax, arrowToken, anonymousMethod.Body);
+            }
+            else
+            {
+                var parameterListSyntax = RemoveType(parameterList)
+                    .WithTrailingTrivia(parameterList.GetTrailingTrivia().WithoutTrailingWhitespace().Add(SyntaxFactory.ElasticSpace));
+                lambdaExpression = SyntaxFactory.ParenthesizedLambdaExpression(anonymousMethod.AsyncKeyword, parameterListSyntax, arrowToken, anonymousMethod.Body);
+            }
+
+            return lambdaExpression
+                .WithAdditionalAnnotations(Formatter.Annotation);
+        }
+
+        private static ParameterListSyntax RemoveType(ParameterListSyntax parameterList)
+        {
+            return parameterList.WithParameters(SyntaxFactory.SeparatedList(parameterList.Parameters.Select(x => RemoveType(x)), parameterList.Parameters.GetSeparators()));
+        }
+
+        private static ParameterSyntax RemoveType(ParameterSyntax parameterSyntax)
+        {
+            var syntax = parameterSyntax.WithType(null)
+                .WithTrailingTrivia(parameterSyntax.Type.GetLeadingTrivia().Concat(parameterSyntax.DescendantTrivia()));
```

Comment:
:question: I would have expected `WithLeadingTrivia` here, why are you using `WithTrailingTrivia`?